vmdb/app/models/service_template_orchestration.rb

```diff
@@ -1,6 +1,16 @@
 class ServiceTemplateOrchestration < ServiceTemplate
   include ServiceOrchestrationMixin
 
+  before_save :remove_invalid_resource
+
+  def remove_invalid_resource
+    service_resources.each { |r| r.delete unless r.resource(true) }
```

Comment:
You do not want to delete a instance in a collection you are looping over.  Use `delete_if` instead of each.
```ruby
service_resources.delete_if { |r| r.resource(true).nil? }
```

Also, is the `true` really required here?